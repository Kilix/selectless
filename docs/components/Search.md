# Search
The default component is an `<input type='text'>`.

| property        | type/description                                              | default |
|-----------------|---------------------------------------------------------------|---------|
| `caseSensitive` | `boolean` - determine if the search is caseSensitive or not   |  false  |
| `disabled`      | `boolean` - return if the select is disabled                  |    -    |

Search's render receive a lot of props :

| property           | type       | description                                         |
|--------------------|------------|-----------------------------------------------------|
| `caseSensitive`    | `boolean`  | -                                                   |
| `clearSearchValue` | `function` | -                                                   |
| `clearValue`       | `function` | -                                                   |
| `onChange`         | `function` | expect the event from the input target as parameter |
| `searchValue`      | `function` | -                                                   |
| `toggleSearch`     | `function` | -                                                   |
| `toggleSelect`     | `function` | -                                                   |
| `value`            | `boolean`  | -                                                   |