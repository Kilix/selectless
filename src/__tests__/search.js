/* global it, expect */

import React from 'react'
import renderer from 'react-test-renderer'

import { Select } from '../'
import Search from '../components/search'
const options = [
  { label: 'Paris', value: 'paris' },
  { label: 'Tokyo', value: 'tokyo' }
]
it('Search', () => {
  const tree = renderer
    .create(
      <Select name='search' options={options}>
        <Search />
      </Select>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
