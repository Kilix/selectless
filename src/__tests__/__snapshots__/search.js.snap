// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search 1`] = `
<div>
  <input
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;
